### PR TITLE
feat: PickSpectrum

### LINKED ISSUE #11: SpectrumExtractor::PickSpectrum
pick peaks in a spectrum.  This function will be modeled closely off of the PeakPickerMRM class (i.e., first filter using a gausian or svgolay filter and then pick using PeakPickerHiRes).  The function call order to ExtractSpectrum will then be 1. PickSpectrum, 2. AnnotateSpectrum, 3. ScoreSpectrum, and 4. MatchSpectrum.

### PR DESCRIPTION
**Pull request by pcolaianni**
closes #11 
The first commit has already been reviewed in another PR. Unfortunately I couldn't make a better comparison so I had to use develop branch again.
So basically what you may want to review is this: https://github.com/biosustain/OpenMS/pull/13/files/9f57a1f93b0093117a661f04049035201af6c4d7..8770c820dba33726bb3f156a6831f8eca9c70298

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h [hunk 1]
```diff
@@ -56,26 +61,43 @@ namespace OpenMS
     SpectrumExtractor();
     virtual ~SpectrumExtractor();
 
-    void setRTWindow(double rt_window);
+    void setRTWindow(const double& rt_window);
     double getRTWindow() const;
 
-    void setMinScore(double min_score);
+    void setMinScore(const double& min_score);
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h]: Sorry I forgot to mention the const in front of the parameter.

File: src/openms/source/ANALYSIS/OPENSWATH/SpectrumExtractor.cpp [hunk 2]
```diff
@@ -115,18 +165,93 @@ namespace OpenMS
...
+    params.setMaxFloat("min_reverse_match", 1.0);
+
+    params.setValue("mz_tolerance", 0.1, "Mass to Charge tolerance.");
+
+    params.setValue("mz_tolerance_units", "Da", "Mass to Charge tolerance units.");
     params.setValidStrings("mz_tolerance_units", ListUtils::create<String>("ppm,Da"));
+
+    params.setValue("sgolay_frame_length", 15, "The number of subsequent data points used for smoothing.\nThis number has to be uneven. If it is not, 1 will be added.");
+    params.setValue("sgolay_polynomial_order", 3, "Order of the polynomial that is fitted.");
+    params.setValue("gauss_width", 0.2, "Gaussian width in Da, estimated peak size.");
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/SpectrumExtractor.cpp]: "Gaussian width in Da or ppm..."

File: src/tests/class_tests/openms/source/SpectrumExtractor_test.cpp [hunk 3]
```diff
@@ -348,7 +408,38 @@ END_SECTION
...
 START_SECTION(getParameters().getDescription("rt_window"))
 {
-  TEST_EQUAL(ptr->getParameters().getDescription("rt_window"), "Retention time window.")
+  TEST_EQUAL(ptr->getParameters().getDescription("rt_window"), "Retention time window in seconds.")
+}
+END_SECTION
+
+START_SECTION(pickSpectrum())
+{
+  // TODO replace and improve this test
```
Comment (dmccloskey) [inline src/tests/class_tests/openms/source/SpectrumExtractor_test.cpp]: You could add in a test for the m/z and intensity for the identified peaks.

**Review by dmccloskey — COMMENTED** [review]
Looks good.  I added a comment on a potential test as an alternative to the output script.

File: src/tests/class_tests/openms/source/SpectrumExtractor_test.cpp (hunk 3 above)
Comment (pcolaianni) [reply src/tests/class_tests/openms/source/SpectrumExtractor_test.cpp]: created an issue for this: https://github.com/biosustain/OpenMS/issues/14

**Comment by pcolaianni** [comment]
"merged" in https://github.com/biosustain/OpenMS/commit/fbae18adb039cc7cb3f6479152172dc3bf29824b

**Pull request closed without merging**